From 9454c91777b33cab8ab3b335918fb2fb25468357 Mon Sep 17 00:00:00 2001
From: "Mazlan, Hazwan Arif" <hazwan.arif.mazlan@intel.com>
Date: Sun, 20 Mar 2022 21:52:34 +0800
Subject: [PATCH xserver 1/6] dri2: Add support for TGL

In addition, have iris be chosen as the DRI driver for TGL platforms.

Signed-off-by: Vivek Kasireddy <vivek.kasireddy@intel.com>
Signed-off-by: Mazlan, Hazwan Arif <hazwan.arif.mazlan@intel.com>
---
 hw/xfree86/dri2/pci_ids/iris_pci_ids.h      | 12 ++++++++++++
 hw/xfree86/dri2/pci_ids/pci_id_driver_map.h |  9 ++++++++-
 2 files changed, 20 insertions(+), 1 deletion(-)
 create mode 100644 hw/xfree86/dri2/pci_ids/iris_pci_ids.h

diff --git a/hw/xfree86/dri2/pci_ids/iris_pci_ids.h b/hw/xfree86/dri2/pci_ids/iris_pci_ids.h
new file mode 100644
index 000000000000..cb4c6dc32c48
--- /dev/null
+++ b/hw/xfree86/dri2/pci_ids/iris_pci_ids.h
@@ -0,0 +1,12 @@
+CHIPSET(0x9A40, tgl_gt2, "Intel(R) Xe Graphics TGL GT2")
+CHIPSET(0x9A49, tgl_gt2, "Intel(R) Xe Graphics TGL GT2")
+CHIPSET(0x9A59, tgl_gt2, "Intel(R) Graphics TGL GT2")
+CHIPSET(0x9A60, tgl_gt1, "Intel(R) UHD Graphics TGL GT1")
+CHIPSET(0x9A68, tgl_gt1, "Intel(R) UHD Graphics TGL GT1")
+CHIPSET(0x9A70, tgl_gt1, "Intel(R) UHD Graphics TGL GT1")
+CHIPSET(0x9A78, tgl_gt2, "Intel(R) UHD Graphics TGL GT2")
+CHIPSET(0x9AC0, tgl_gt2, "Intel(R) UHD Graphics TGL GT2")
+CHIPSET(0x9AC9, tgl_gt2, "Intel(R) UHD Graphics TGL GT2")
+CHIPSET(0x9AD9, tgl_gt2, "Intel(R) UHD Graphics TGL GT2")
+CHIPSET(0x9AF8, tgl_gt2, "Intel(R) UHD Graphics TGL GT2")
+
diff --git a/hw/xfree86/dri2/pci_ids/pci_id_driver_map.h b/hw/xfree86/dri2/pci_ids/pci_id_driver_map.h
index 04f372279034..8b864e2c12ef 100644
--- a/hw/xfree86/dri2/pci_ids/pci_id_driver_map.h
+++ b/hw/xfree86/dri2/pci_ids/pci_id_driver_map.h
@@ -19,6 +19,13 @@ static const int i965_chip_ids[] = {
 #undef CHIPSET
 };
 
+static const int iris_chip_ids[] = {
+#define CHIPSET(chip, family, name) chip,
+#include "pci_ids/iris_pci_ids.h"
+#undef CHIPSET
+};
+
+
 #ifndef DRIVER_MAP_GALLIUM_ONLY
 static const int r100_chip_ids[] = {
 #define CHIPSET(chip, name, family) chip,
@@ -65,7 +72,7 @@ static const struct {
 } driver_map[] = {
    { 0x8086, "i915", i915_chip_ids, ARRAY_SIZE(i915_chip_ids) },
    { 0x8086, "i965", i965_chip_ids, ARRAY_SIZE(i965_chip_ids) },
-   { 0x8086, "i965", NULL, -1 },
+   { 0x8086, "iris", iris_chip_ids, ARRAY_SIZE(iris_chip_ids) },
 #ifndef DRIVER_MAP_GALLIUM_ONLY
    { 0x1002, "radeon", r100_chip_ids, ARRAY_SIZE(r100_chip_ids) },
    { 0x1002, "r200", r200_chip_ids, ARRAY_SIZE(r200_chip_ids) },
-- 
2.34.1

